### DISCUSSION TITLE
Is there a maximum number of channels per dongle in multiscan?

### DISCUSSION THREAD

**Discussion by Moguldan** [rtl-airband/RTLSDR-Airband#401] [Q&A]
Here is my question. Is there eight still the maximum number of channels in multiscan mode, assuming they are all within the bandwidth of your RTL dongle? 

Also, I want to thank everyone who helped develop this. Being a Linux noob, it took me quite a bit to figure out, but the documentation here has been so good even I could figure it out, with a lot of time spent reading what you have done. THANK YOU! 

I don't see a maximum listed anywhere in the wiki, but I did see several 8 in the example config file at https://github.com/n7ipb/RTLSDR-Airband/blob/master/rtl_airband.conf.example, but this is seven years old. This seems to be managed as a separate page. Is that 8 max still accurate? If not, when did it change?

**Comment by szpajder** [comment]
What you probably mean is multichannel mode.

The number of channels in this mode is unlimited. The 8-channel limit has been removed in version 3.0.0.

**Comment by Moguldan** [comment]
What is the maximum number of channels that can be mixed? I currently have 38 I seemed to be ok when I only had 26.  Getting error "Configuration error: devices.[0] channels.[32] outputs[0]: could not connect to mixer my_mixer1: too many inputs"

I would copy and paste my config file but obviously, it will be very long.

**Comment by Moguldan** [comment]
Tried eliminating channels until it worked. seems to be 32 at least on version 3.1.0. It does seem to be working. Does anyone know if this is the limit with newer versions?

I am using this for Marine VHF. Decided on my channels by taking the FCC list of frequency, Eliminating the duplex marine operators (not really used much anymore), and non-audio stations (for example channels 70 87 88, and so on), I think I have a count of 35 channels.  Debating which three I don't care about. And double checking my own list. I am more than likely covering a lot of channels that will never be used but as there are a lot of untrained users they tend to pop up on weird channels you don't expect from time to time.

**Comment by charlie-foxtrot** [comment]
Yes, the maximum number of channels that can be included into a single mixer is 32.

That value could be increased, but seems really high as it is given the channels will step on each other if transmitting at the same time.

What are you doing with the audio output?  you could have one mixer for the "common" channels and a second mixer for the "uncommon" channels

**Comment by charlie-foxtrot** [comment]
@cdknox has a PR open #396 to restructure the code to support larger values of `MAX_MIXINPUTS` but as of now that PR will not change it from 32

**Comment by Moguldan** [comment]
Thank you Charlie-foxtrot. I think I understand the current fix is to make two mixers and then mix them together. My setup currently works well with the 32 channels and for a Brodcastricy Feed.

I am doing marine VHF radios. (Ship and boat traffic) There are many channels an operator can choose to use within the range of one RTL Dongle. Currently, it's unlikely that more than one or two will actually be having traffic at a time, but all channels can potentially have traffic. Most radio traffic starts on channel 16, or channel 9 then operators switch to whatever they feel like. There are conventions, but many recreational boaters need to learn them, and in practice, they just pick another channel at random.

There are channels that are unlikely to have traffic, though. None of the duplex frequencies have repeaters around here, for instance. Some are limited to digital traffic. It still adds up to more than 32 potential voice channels. See: [](https://copradar.com/marine/channels_us.pdf) or [](https://wiki.radioreference.com/index.php/Marine_VHF_Channels)

**Reply by charlie-foxtrot** [reply]
Thanks for the context. Mixers can not feed mixers, so using two mixers would mean listening to both audio outputs simultaneously if you want to hear everything.

Given the valid use case I'll make changes to support unlimited mixer inputs.

**Comment by charlie-foxtrot** [comment]
mixers with unlimited input count has been merged to `unstable`
